src/Compiler/Microsoft.AspNetCore.Razor.Language/test/DefaultRazorTagHelperBinderPhaseTest.cs

```diff
@@ -1335,8 +1335,10 @@ public void ComponentDirectiveVisitor_DoesNotMatchForUsingAliasAndStaticUsings()
     [InlineData("", "", true)]
     [InlineData("Foo", "Project", true)]
     [InlineData("Project.Foo", "Project", true)]
+    [InlineData("Project.Foo", "global::Project", true)]
     [InlineData("Project.Bar.Foo", "Project.Bar", true)]
     [InlineData("Project.Foo", "Project.Bar", false)]
+    [InlineData("Project.Foo", "global::Project.Bar", false)]
     [InlineData("Project.Bar.Foo", "Project", false)]
     [InlineData("Bar.Foo", "Project", false)]
     public void IsTypeInNamespace_WorksAsExpected(string typeName, string @namespace, bool expected)
```

Comment:
💡 `expected` shouldn't be a parameter here. It should be a variable inside the test which is initialized with a switch expression, e.g.:

```
var expected = (typeName, @namespace) switch
{
  ("", "") => true,
  ("Foo", "Project") => true,
  ("Project.Foo", "Project") => true,
  ("Project.Foo", "global::Project") => true,
  _ => false,
};
```

Reply:
Why would that be better? It would need duplicating the test data.

Reply:
> Why would that be better?

The parameters to a test are part of the test identity. When expected outcomes are part of the parameter list, it is impossible to update code in a manner that changes the expected outcome of a test without deleting the entire test history.

> It would need duplicating the test data.

Yes, and for this type of situation it's completely acceptable. Duplication is not as much a maintenance problem when coupled (both copies are highly likely to appear on the same screen at the same time).

Reply:
That's interesting. I wouldn't think expected outcomes of unit tests change, that's what snapshot tests are for. Although I guess the line between the two is not that clear.